app/src/main/java/io/github/zadam/triliumsender/SendNoteActivity.kt

```diff
@@ -51,6 +67,50 @@ class SendNoteActivity : AppCompatActivity() {
         }
     }
 
+    /**
+     * If we're the target of a SEND intent, pre-fill the note's contents.
+     *
+     * @param intent, the Intent object, already verified to be of type text/plain.
+     */
+    private fun handleSendText(intent: Intent) {
+        val tag = "SendNoteHandleSendText"
+
+        // Many apps will suggest a "Text Subject", as in an email.
+        val suggestedSubject = intent.getStringExtra((Intent.EXTRA_SUBJECT))
+        if (suggestedSubject != null && suggestedSubject.isNotEmpty()) {
+            // Use suggested subject.
+            noteTitleEditText.setText(suggestedSubject, TextView.BufferType.EDITABLE)
+        } else {
+            // Try to craft a sane default title.
+            var referrerName = "Android"
+
+            // SDKs greater than 23 can access the referrer's package URI...
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP_MR1) {
+                val referrer = this.referrer
+                if (referrer != null && referrer.host != null) {
+                    // Which we can use to get an AppInfo handle...
+                    try {
+                        // Which we can use to get an app label!
+                        val referrerInfo = packageManager.getApplicationInfo(referrer.host.toString(), 0)
+                        val potentialReferrerName = referrerInfo.loadLabel(packageManager).toString()
+                        // Sanity check: is it an empty string?
+                        if (potentialReferrerName.isNotEmpty()) {
+                            referrerName = potentialReferrerName
+                        }
+                    } catch (e: PackageManager.NameNotFoundException) {
+                        // Oh well, we have the default name to fall back on.
+                        Log.w(tag, "Could not find package label for package name ${referrer.host}")
+                    }
+                }
+            }
+            // Ultimately, set the note title.
+            noteTitleEditText.setText(getString(R.string.share_note_title, referrerName), TextView.BufferType.EDITABLE)
+        }
+        // And populate the note body!
+        noteContentEditText.setText(intent.getStringExtra(Intent.EXTRA_TEXT), TextView.BufferType.EDITABLE)
+    }
```

Comment:
This function is not the most kotlin-ish in the world. Happy to rewrite any parts that are particularly offensive to you.